tools/jsrun.py

```diff
@@ -45,12 +45,20 @@ def make_command(filename, engine=None, args=[]):
   # Use "'d8' in" because the name can vary, e.g. d8_g, d8, etc.
   is_d8 = 'd8' in jsengine or 'v8' in jsengine
   is_jsc = 'jsc' in jsengine
+  is_wasmer = 'wasmer' in jsengine
+  is_wasmtime = 'wasmtime' in jsengine
   # Disable true async compilation (async apis will in fact be synchronous) for now
   # due to https://bugs.chromium.org/p/v8/issues/detail?id=6263
   shell_option_flags = ['--no-wasm-async-compilation'] if is_d8 else []
+  command_flags = []
+  if is_wasmer:
+    command_flags += ['run']
+  if is_wasmer or is_wasmtime:
+    # in a wasm runtime, run the wasm, not the js
+    filename = filename[:-3] + '.wasm'
```

Comment:
So we are still geneating the .js file in all cases?  But then running the wasm on its own at this late stage?

I'd kind of rather push this logic back in the `do_run` method of the test runner, but I'm OK with cleaning this up later if you prefer?

Reply:
We still build js+wasm normally. We then test the js, and separately also test the wasm without the js. Basically, when we ask to run the app in a js vm we run the js, and in wasm vm the wasm.

I actually think this makes sense to do here. This was we can call `run` from anywhere, and if the vm we run in happens to be a wasm vm, it will just work. That may be useful in `other`, `browser`, etc. eventually, and not just in `do_run` which is mainly for `core`.